File: Day02/go/part02.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"strconv"
	"strings"
)

func passwordValidator() int {

	f, err := os.Open("./blob.txt")
	if err != nil {
		log.Fatal(err)
	}
	defer f.Close()

	scanner := bufio.NewScanner(f)
	total := 0
	for scanner.Scan() {
		text := strings.Fields(scanner.Text())
		params := text[0]
		firstIndex, err := strconv.Atoi(strings.Split(params, "-")[0])
		secondIndex, err := strconv.Atoi(strings.Split(params, "-")[1])
		char := strings.Replace(text[1], ":", "", -1)
		password := text[len(text)-1]
		runes := []rune(password)
		if char == string(runes[firstIndex-1]) {
			if string(runes[firstIndex-1]) != string(runes[secondIndex-1]) {
				total++
			}
		}
		if char == string(runes[secondIndex-1]) {
			if string(runes[firstIndex-1]) != string(runes[secondIndex-1]) {
				total++
			}
		}

		if err != nil {
			log.Fatal(err)
		}
	}
	return total
}

func main() {
	fmt.Println(passwordValidator())
}

File: Day02/part02.py
import os

file = os.path.join(os.path.dirname(__file__), "blob.txt")


def file_reader(file: str) -> str:
    with open(file, "r") as f:
        return f.read()


def solver(s: str) -> int:
    total = 0
    for line in s.splitlines():
        ln = line.split()
        params = ln[0]
        first_index = int(params.split("-")[0]) - 1
        second_index = int(params.split("-")[-1]) - 1
        char = ln[1].replace(":", "")
        password = ln[-1]
        if (char == password[first_index]) ^ (char == password[second_index]):
            total += 1
    return total


if __name__ == "__main__":
    print(solver(file_reader(file)))
